packages/foreman/rubygem-ffi/rubygem-ffi.spec

```diff
@@ -70,6 +70,16 @@ cp -a .%{gem_extdir_mri}/{gem.build_complete,*.so} %{buildroot}%{gem_extdir_mri}
 # Prevent dangling symlink in -debuginfo (rhbz#878863).
 rm -rf %{buildroot}%{gem_instdir}/ext/
 
+%check
+%{?scl:scl enable %{scl} - << \EOF}
+mkdir gem_ext_test
+cp -a %{buildroot}%{gem_dir} gem_ext_test/
+mkdir -p gem_ext_test/gems/extensions/%{_arch}-%{_target_os}/$(ruby -r rbconfig -e 'print RbConfig::CONFIG["ruby_version"]')/
+cp -a %{buildroot}%{gem_extdir_mri} gem_ext_test/gems/extensions/%{_arch}-%{_target_os}/$(ruby -r rbconfig -e 'print RbConfig::CONFIG["ruby_version"]')/
+GEM_PATH="./gem_ext_test/gems:$GEM_PATH" ruby -e "require '%{gem_name}'"
+rm -rf gem_ext_test
```

Comment:
That's such a pretty hammer! I suppose I have to ask:

 * Can we macro this ?
 * Or just add it to gem2rpm to get included in native library gems?

Reply:
I will not try to break our macros twice in a week.

That said, gem2rpm template is what @ekohl did in his original check code and so I would just update that.

Reply:
Hah, ok. I would maybe nominate this if we turned it into a macro to be its own special package just for build macros not tied to building foreman itself.

Reply:
In my experience with https://github.com/theforeman/foreman-packaging/pull/4921 it turns out that a lot of gems aren't a `require '%{gem_name}'` but actually a different name.

@evgeni is this something you'd like to take over? I'm getting the feeling that I've opened Pandora's box with it and won't have time to finish.

Reply:
I like boxes. You can stack them.

Yes, I could take that one over as part of "make it el8"